internal/testutil/test_schemas.go

```diff
@@ -0,0 +1,400 @@
+package testutil
+
+// Patterns:
+// Resource schema for required fields only
```

Comment:
Go has a built-in template language (https://pkg.go.dev/text/template). It would allow us to have one resource template as it can handle optional arguments (unlike the `fmt` package functions). As a bonus, we could use variable names in the template content and input. So `packetfabric_cloud_router` would look like this:

```go
const ResourcePacketfabricCloudRouterTemplate = `resource "packetfabric_cloud_router" "{{ .ResourceName }}" {
	name          = "{{ .Name }}"
        account_uuid  = "{{ .AccountUUID }}"
	capacity      = "{{ .Capacity }}"
}`

t := template.Must(template.New("resource").Parse(ResourcePacketfabricCloudRouterTemplate))
data := struct {
	ResourceName string
	Name string
	AccountUUID string
	Capacity string
}{
	"xxx",
	"yyy",
	"zzz",
	"vvv",
}
t.Execute(os.Stdout, data)
```

The main pain in the ass in current tests was (at least for me) that I had to watch for the order of arguments passed to function, as it was easy to make a mistake.

Reply:
Yes, I was considering that too, but that would make the schema creation very painful, because we'll need to recreate the struct order every time we want to create a new test. I think using `fmt` will give us that flexibility back and gonna have the same behavior.

Reply:
Well, with templates, you can handle optional inputs with `if` conditional. For `fmt`, you must create a separate schema for optional values. Of course, an anonymous structure is not the only way of providing input, we can define a reusable structure or use maps.

Reply:
The choice between `template` and `fmt` balances the tradeoff of a more complicated initial setup to have a lot of flexibility later. That would be beneficial if we wanted to have a lot of different (by a number of inputs) tests for one resource. If we don't plan such a thing, I agree, we can go with `fmt`.

Reply:
Resolving this conversation. Thanks for the comments guys.